File: python/1759.count-number-of-homogenous-substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        total = 0
        MOD = 10**9 + 7
        for _, g in itertools.groupby(s):
            s = len(list(g))
            total = (total + s * (s + 1) // 2) % MOD
        return total

File: golang/1759.count-number-of-homogenous-substrings.go
const MOD = int(1e9) + 7

func countHomogenous(s string) int {
    total := 0
    var i, j int
    for i = 0; i < len(s); {
        for j = i + 1; j < len(s) && s[i] == s[j]; j++ {}
        total += (j - i) * (j - i + 1) / 2
        i = j
    }

    return total % MOD
}